src/app/services/actions/query-action-creators.ts

```diff
@@ -31,9 +32,20 @@ export function runQuery(query: IQuery): Function {
       }
     }
 
-    const headers = { Authorization: `Bearer ${authToken}` };
+    const headers = {
+      'Authorization': `Bearer ${authToken}`,
+      'Content-Type': 'application/json',
+    };
 
-    return fetch(graphUrl, { headers })
+    let options: IRequestOptions = {};
+    options = { method: query.selectedVerb, headers};
+
+    if (query.sampleBody && query.selectedVerb !== 'GET') {
```

Comment:
To improve readability you can make the expressions in the condition statement variables.
```
const hasBody = !!query.sampleBody // !! forces it to be true or false
const isGET = query.selectedVerb

if (hasBody && !isGET) {
  ...
}
```

Reply:
Should this work for `DELETE` requests